src/main/java/edu/harvard/iq/dataverse/export/HtmlCodeBookExporter.java

```diff
@@ -0,0 +1,84 @@
+package edu.harvard.iq.dataverse.export;
+
+import com.google.auto.service.AutoService;
+import edu.harvard.iq.dataverse.DatasetVersion;
+import edu.harvard.iq.dataverse.export.ddi.DdiExportUtil;
+import edu.harvard.iq.dataverse.export.spi.Exporter;
+import edu.harvard.iq.dataverse.util.BundleUtil;
+
+import javax.json.JsonObject;
+import javax.ws.rs.core.MediaType;
+import javax.xml.stream.XMLStreamException;
+import java.io.File;
+import java.io.OutputStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+@AutoService(Exporter.class)
+public class HtmlCodeBookExporter implements Exporter {
+
+    @Override
+    public String getProviderName() {
+        return "html";
+    }
+
+    @Override
+    public String getDisplayName() {
+        return  BundleUtil.getStringFromBundle("dataset.exportBtn.itemLabel.html") != null ? BundleUtil.getStringFromBundle("dataset.exportBtn.itemLabel.html") : "DDI html codebook";
+    }
+
+    @Override
+    public void exportDataset(DatasetVersion version, JsonObject json, OutputStream outputStream) throws ExportException {
+        try {
+            Path cachedMetadataFilePath = Paths.get(version.getDataset().getFileSystemDirectory().toString(), "export_ddi" + ".cached");
```

Comment:
getFileSystemDirectory makes me suspect that this might not work on S3 or Swift.

Reply:
@pdurbin You are probably right. I did not test it for S3 or Swift, it probably will not work with it. That is why I just replaced getFileSystemDirectory  with getExport from ExportService. As I understand it should work for S3 and Swift.

Reply:
@lubitchv perfect, thanks! I just moved this to QA.